e2e_deployment_files/edgehub_restart_deployment.template.json

```diff
@@ -0,0 +1,230 @@
+            "type": "docker",
+            "status": "running",
+            "restartPolicy": "always",
+            "env": {
+              "Logging:LogLevel:Microsoft": {
+                "value": "Error"
+              },
+              "trackingId": {
+                "value": "<TrackingId>"
+              },
+              "testStartDelay": {
+                "value": "<TestStartDelay>"
+              },
+              "testDuration": {
+                "value": "<TestDuration>"
+              },
+              "verificationDelay": {
+                "value": "<TestResultCoordinator.VerificationDelay>"
+              },
+              "eventHubConnectionString": {
+                "value": "<TestResultCoordinator.EventHubConnectionString>"
+              },
+              "ConsumerGroupName": {
+                "value": "<TestResultCoordinator.ConsumerGroupId>"
+              },
+              "optimizeForPerformance": {
+                "value": "<TestResultCoordinator.OptimizeForPerformance>"
+              },
+              "logAnalyticsWorkspaceId": {
+                "value": "<LogAnalyticsWorkspaceId>"
+              },
+              "logAnalyticsSharedKey": {
+                "value": "<LogAnalyticsSharedKey>"
+              },
+              "logAnalyticsLogType": {
+                "value": "<TestResultCoordinator.LogAnalyticsLogType>"
+              },
+              "IOT_HUB_CONNECTION_STRING": {
+                "value": "<IoTHubConnectionString>"
+              }
+            },
+            "settings": {
+              "image": "<Container_Registry>/microsoft/azureiotedge-test-result-coordinator:<Build.BuildNumber>-linux-<Architecture>",
+              "createOptions": "{\"HostConfig\": {\"PortBindings\": {\"5001/tcp\": [{\"HostPort\": \"5001\"}]}}}"
+            }
+          }
+        }
+      }
+    },
+    "$edgeHub": {
+      "properties.desired": {
+        "schemaVersion": "1.0",
+        "routes": {
+          "edgeHubRestarterToRelayer1": "FROM /messages/modules/loadGen1/outputs/output1 INTO BrokeredEndpoint(\"/modules/relayer1/inputs/input1\")"
+        },
+        "storeAndForwardConfiguration": {
+          "timeToLiveSecs": 86400
+        }
+      }
+    },
+    "testResultCoordinator": {
+      "properties.desired": {
+        "schemaVersion": "1.0",
+        "reportMetadataList": {
+          "reportMetadata1": {
+            "TestReportType": "EdgeHubRestartDirectMethodResult",
+            "TestOperationResultType": "EdgeHubRestartDirectMethod",
+            "SenderSource": "edgeHubRestartTester1.EdgeHubRestartDirectMethod",
+            "ReceiverSource": "directMethodReceiver1.receive"
+          },
+          "reportMetadata2": {
+            "TestReportType": "EdgeHubRestartMessageResult",
+            "TestOperationResultType": "EdgeHubRestartMessage",
+            "SenderSource": "edgeHubRestartTester1.EdgeHubRestartMessage",
+            "ReceiverSource": "relayer1.receive"
+          }
```

Comment:
you need to add network controller module and add network controller report as well.

Reply:
The offline scenario is not handled in this PR.